modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/api.ts

```diff
@@ -63,7 +63,21 @@ function getValidFields({
 		const type = propertyValue.type;
 
 		if (type === EFieldType.ARRAY) {
-			if (propertyValue.items && propertyValue.items.$ref) {
+			if (propertyValue.items?.$ref) {
+				fields.push({
+					children: getValidFields({
+						contextPath: `${contextPath}${propertyKey}${FDS_NESTED_FIELD_NAME_DELIMITER}`,
+						schemaName: propertyValue.items.$ref.replace(
+							/^.*\//,
+							''
+						),
+						schemas,
+					}),
+					label: propertyKey,
+					name: `${contextPath}${propertyKey}${FDS_NESTED_FIELD_NAME_PARENT_SUFFIX}`,
+					type: type ? type : 'object',
```

Comment:
we do know `type === EFieldType.ARRAY` here so we should not need this check.

Reply:
Scalar arrays do not need this process. They go through [L124](https://github.com/liferay-frontend/liferay-portal/pull/4135/files#diff-8b819d07f50b442d5ec220fd20383c40b9713e088f455625482a2c6d9b8e9a3dR124), default processing.